File: src/main/java/org/leetcode/EvaluateReversePolishNotation_150.java
package org.leetcode;

import java.util.Stack;

public class EvaluateReversePolishNotation_150 {
    public int evalRPN(String[] tokens) {
        Stack<Integer> stack = new Stack<>();
        for(String s: tokens){

            if(s.equals("+")){
                stack.push(stack.pop() + stack.pop());
            } else if(s.equals("-")){
                Integer n = stack.pop();
                stack.push(stack.pop() - n);
            } else if(s.equals("*")){
                stack.push(stack.pop() * stack.pop());
            } else if(s.equals("/")){
                Integer n = stack.pop();
                stack.push(stack.pop() / n);
            } else{
                stack.push(Integer.valueOf(s));
            }
        }
        return stack.pop();
    }
}

File: src/main/python/EvaluateReversePolishNotation_150.py
from typing import List


class Solution:
    def evalRPN(self, tokens: List[str]) -> int:
        stack = []
        for c in tokens:
            if stack and c == '+':
                n1 = int(stack.pop())
                stack.append(int(stack.pop()) + n1)
            elif stack and c == '-':
                n1 = int(stack.pop())
                stack.append(int(stack.pop()) - n1)
            elif stack and c == '*':
                n1 = int(stack.pop())
                stack.append(int(stack.pop()) * n1)
            elif stack and c == '/':
                n1 = int(stack.pop())
                stack.append(int(stack.pop()) / n1)
            else:
                stack.append(c)
        return int(stack[0])